server/src/modules/task/task.service.ts

```diff
@@ -1,17 +1,42 @@
-import { Injectable } from '@nestjs/common';
-import { User } from '../user/user.model';
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { Task } from './task.model';
+import { InjectModel } from '@nestjs/sequelize';
+import { TaskDTO as TaskInterface } from './task.interface';
 
 @Injectable()
 export class TaskService {
-  async createUser(createUserDto: any): Promise<User> {
-    const user = new User();
-    user.firstName = createUserDto.username;
-    user.email = createUserDto.email;
+  constructor(
+    @InjectModel(Task)
+    private readonly taskModel: typeof Task,
+  ) {}
 
-    return user.save();
+  async createTask(taskData: Task): Promise<Task> {
+    return this.taskModel.create(taskData);
   }
 
-  async findAllUsers(): Promise<User[]> {
-    return User.findAll();
+  async getAllTask(): Promise<TaskInterface[]> {
+    return this.taskModel.findAll();
+  }
+
+  async getTaskById(id: number): Promise<Task> {
+    const task = await this.taskModel.findByPk(id);
+    if (!task) {
+      throw new NotFoundException(`Task with id ${id} not found`);
+    }
+    return task;
+  }
+
+  async updateTask(id: number, taskData: Task): Promise<Task> {
```

Comment:
I actually just made the same comment on Aidai and FuMing's code: you want to make sure you don't forget to add authorization here later. Currently, any user can edit ANY task (same for delete).

One possibility would be to make `updateTask` take a `user_id` field to represent the current user (and make sure the task belongs to that user), and then after you implement authentication you could replace `user_id` with the logged-in user. Or you could add a task to your task tracker to make sure you don't forget to add simple authorization logic here later (or you could at least add a TODO comment in the code).